packages/redwood/src/index.ts

```diff
@@ -150,10 +160,47 @@ export const build: BuildV2 = async ({
   const apiDistPath = join(workPath, 'api', 'dist', 'functions');
   const webDistPath = join(workPath, 'web', 'dist');
   const lambdaOutputs: { [filePath: string]: Lambda } = {};
-  const staticOutputs = await glob('**', webDistPath);
+
+  // Strip out the .html extensions
+  // And populate staticOutputs map with updated paths and contentType
+  const webDistFiles = await glob('**', webDistPath);
+  const staticOutputs: Record<string, FileFsRef> = {};
+
+  for (const [fileName, fileFsRef] of Object.entries(webDistFiles)) {
+    const parsedPath = parsePath(fileFsRef.fsPath);
+
+    if (parsedPath.ext !== '.html') {
+      // No need to transform non-html files
+      staticOutputs[fileName] = fileFsRef;
+    } else {
+      const fileNameWithoutExtension = pathBasename(fileName, '.html');
+
+      const pathWithoutHtmlExtension = join(
+        parsedPath.dir,
+        fileNameWithoutExtension
+      );
+
+      renameSync(fileFsRef.fsPath, pathWithoutHtmlExtension);
+      fileFsRef.contentType = 'text/html; charset=utf-8';
+      fileFsRef.fsPath = pathWithoutHtmlExtension;
+
+      // @NOTE: Filename is relative to webDistPath
+      staticOutputs[relative(webDistPath, pathWithoutHtmlExtension)] =
+        fileFsRef;
+    }
+  }
 
   // Each file in the `functions` dir will become a lambda
-  const functionFiles = await glob('*.js', apiDistPath);
+  // Also supports nested functions like:
+  // ├── functions
+  // │   ├── bazinga
+  // │   │   ├── bazinga.js
+  // │   ├── graphql.js
+
+  const functionFiles = {
+    ...(await glob('*.js', apiDistPath)), // top-level
+    ...(await glob('*/*.js', apiDistPath)), // one-level deep
```

Comment:
Can you use a single `**/*.js` instead of two globs?

Reply:
This is intentional, suggested by Nate. We only want to go two levels deep - because that is the "most compatible" option  e.g. Netlify behaves the same way, and I'm also changing our api server to also adopt this behaviour. Previously it was one level deep, because of Vercel - now we're good everywhere :) 

In Redwood typically we use the "deep" option 
```
export const findApiDistFunctions = (cwd: string = getPaths().api.base) => {
  return fg.sync('dist/functions/**/*.{ts,js}', {
    cwd,
    deep: 2, // 👈 We don't support deeply nested api functions, to maximise compatibility with deployment providers
    absolute: true,
  })
}
```
But the glob func from buildTools doesn't support this, which is why theres the two globs!